store/tikv/region_cache.go

```diff
@@ -365,6 +379,12 @@ func (c *RegionCache) GetTiKVRPCContext(bo *Backoffer, id RegionVerID, replicaRe
 		return nil, nil
 	}
 
+	if cachedRegion.needReload() {
```

Comment:
Why add it?

Reply:
In function `RegionRequestSender.SendReqCtx` there's a retry loop, in which it gets a RPCContext, send a request, and retry if sending failed. If it has tried all peer, it will mark the region as need reload, but reloading won't happen in this loop. So I added it here, and it will cause a fake `EpochNotMatch` error to the invoker of `SendReqCtx`, and the invoker will usually reload the region.

Reply:
🧐 I remembered there's someplace the error is escaped from the loop. Let me check it.

Reply:
When  getRPCContext returns nil, it will cause a empty EpochNotMatch error and excape from the loop..